### PR TITLE
DaService::send_transaction accepts DaData instead of plain Vec<u8>

### PR DESCRIPTION
**Pull request by kpp**
# Description
Instead of sending pre-borshed data to DA we send DaData to bitcoin queue and borsh it inside da service.

## Linked Issues
- Related to https://github.com/chainwayxyz/citrea/issues/972

## Testing
`make test`

### CODE REVIEW

File: crates/sovereign-sdk/adapters/mock-da/src/service.rs [hunk 1]
```diff
@@ -190,22 +192,22 @@ impl MockDaService {
...
-        let _ = self.add_blob(&blocks, &blob, Default::default())?;
+        let _ = self.add_blob(&blocks, blob, Default::default())?;
         Ok(())
     }
 
     fn add_blob(
         &self,
         blocks: &AsyncMutexGuard<'_, DbConnector>,
-        blob: &[u8],
+        blob: Vec<u8>,
```
Comment (rakanalh) [inline crates/sovereign-sdk/adapters/mock-da/src/service.rs:202]: I don't think this was necessary

**Review by rakanalh — APPROVED** [review]
LGTM

File: crates/sovereign-sdk/adapters/mock-da/src/service.rs (hunk 1 above)
Comment (kpp) [reply crates/sovereign-sdk/adapters/mock-da/src/service.rs:202]: It wasn't. but its a mock da, so why not

**Review by eyusufatik — APPROVED** [review]

File: crates/sovereign-sdk/adapters/mock-da/src/service.rs (hunk 1 above)
Comment (eyusufatik) [reply crates/sovereign-sdk/adapters/mock-da/src/service.rs:202]: top kek

**Pull request merged**